### PR TITLE
Fix ESS remote method calls for connect only users

### PR DESCRIPTION
**Pull request by Atlante45**
Fixes a bug that prevents users with connect-only permissions to call methods on server scripts.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8305/app_12719_PR.apk

**Comment by Atlante45** [comment]
## Test Plan:
- Run the first 4 steps of 
[C1932](https://highfidelity.testrail.net/index.php?/cases/view/1932) in your Sandbox
- In your domain server settings, remove your rez permissions so that only your connect permissions remain
- Leave your domain and come back to it
- Attempt to complete the rest of the test plan

**Expected behavior:** The test plan works as expected but will only see the output from the client log:
```[10/24 11:27:03] [DEBUG] [hifi.scriptengine.script] [about:Entities 1] Example.fooDoOnClient(["a","b","1","2","3"]) remoteCallerID:{07164797-3309-4efb-877b-719bd976e5c0}```
You won't see anything in the server log.




[Bug](https://highfidelity.manuscript.com/f/cases/13708/callEntityServerMethod-is-not-called-when-the-client-has-only-Connect-permissions)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8305/HighFidelity-Beta-PR12719-a0c6e1319663ad5855b17f0d30c89c7060a5cabf.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8305/HighFidelity-Beta-PR12719-a0c6e1319663ad5855b17f0d30c89c7060a5cabf.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8307/app_12719_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8307/HighFidelity-Beta-PR12719-de9e1bd30c84153ae1edc2dd604a8e3ed0dc1d50.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8307/HighFidelity-Beta-PR12719-de9e1bd30c84153ae1edc2dd604a8e3ed0dc1d50.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Stopped testing pending further intruction

**Comment by qlimnavarro** [comment]
Currently Testing

File: assignment-client/src/scripts/EntityScriptServer.cpp [hunk 1]
```diff
@@ -81,11 +81,6 @@ EntityScriptServer::EntityScriptServer(ReceivedMessage& message) : ThreadedAssig
                                             this, "handleOctreePacket");
     packetReceiver.registerListener(PacketType::SelectedAudioFormat, this, "handleSelectedAudioFormat");
 
-    auto avatarHashMap = DependencyManager::set<AvatarHashMap>();
-    packetReceiver.registerListener(PacketType::BulkAvatarData, avatarHashMap.data(), "processAvatarDataPacket");
-    packetReceiver.registerListener(PacketType::KillAvatar, avatarHashMap.data(), "processKillAvatar");
-    packetReceiver.registerListener(PacketType::AvatarIdentity, avatarHashMap.data(), "processAvatarIdentityPacket");
```
Comment (birarda) [inline assignment-client/src/scripts/EntityScriptServer.cpp]: Why did we remove these?

**Comment by conklin94122** [comment]
Removing QA Approved pending dev changes. If the changes result in a new build QA will retest (please reapply QA Ready). If no new build is necessary @ me with details, I'll reapply QA Approved and merge.

File: assignment-client/src/scripts/EntityScriptServer.cpp (hunk 1 above)
Comment (Atlante45) [reply assignment-client/src/scripts/EntityScriptServer.cpp]: Dead code - Those functions don't even exist.

Comment (birarda) [reply assignment-client/src/scripts/EntityScriptServer.cpp]: The `AvatarHashMap` they are connected to disagrees with that

Comment (Atlante45) [reply assignment-client/src/scripts/EntityScriptServer.cpp]: nvm - I'm an idiot

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8332/app_12719_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8332/HighFidelity-Beta-PR12719-cb07fc47dc58e86c4982105a04e02e721dfb60db.dmg

**Comment by davidkelly** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8342/app_12719_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8342/HighFidelity-Beta-PR12719-cb07fc47dc58e86c4982105a04e02e721dfb60db.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8342/HighFidelity-Beta-PR12719-cb07fc47dc58e86c4982105a04e02e721dfb60db.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Pull request merged**